src/batch/src/executor/iceberg_scan.rs

```diff
@@ -95,77 +102,28 @@ impl IcebergScanExecutor {
         let data_types = self.schema.data_types();
         let executor_schema_names = self.schema.names();
 
-        let mut eq_delete_file_scan_tasks_map: HashMap<OwnedRow, i64> = HashMap::default();
-        let eq_delete_file_scan_tasks = mem::take(&mut self.eq_delete_file_scan_tasks);
-
-        // Build hash map for equality delete files
-        // Currently, all equality delete files have the same schema which is guaranteed by `IcebergSplitEnumerator`.
-        let mut eq_delete_ids: Option<Vec<_>> = None;
-        for eq_delete_file_scan_task in eq_delete_file_scan_tasks {
-            let mut sequence_number = eq_delete_file_scan_task.sequence_number;
-
-            if eq_delete_ids.is_none() {
-                eq_delete_ids = Some(eq_delete_file_scan_task.project_field_ids.clone());
-            } else {
-                debug_assert_eq!(
-                    eq_delete_ids.as_ref().unwrap(),
-                    &eq_delete_file_scan_task.project_field_ids
-                );
-            }
-
-            let reader = table
-                .reader_builder()
-                .with_batch_size(self.batch_size)
-                .build();
-            let delete_file_scan_stream = tokio_stream::once(Ok(eq_delete_file_scan_task));
-
-            let mut delete_record_batch_stream = reader
-                .read(Box::pin(delete_file_scan_stream))
-                .map_err(BatchError::Iceberg)?;
-
-            while let Some(record_batch) = delete_record_batch_stream.next().await {
-                let record_batch = record_batch.map_err(BatchError::Iceberg)?;
-
-                let chunk = IcebergArrowConvert.chunk_from_record_batch(&record_batch)?;
-                for row in chunk.rows() {
-                    let entry = eq_delete_file_scan_tasks_map
-                        .entry(row.to_owned_row())
-                        .or_default();
-                    *entry = *entry.max(&mut sequence_number);
-                }
-            }
-        }
+        let position_delete_filter = PositionDeleteFilter::new(
+            mem::take(&mut self.position_delete_file_scan_tasks),
+            &table,
+            self.batch_size,
+        )
```

Comment:
We can pass data_file_scan_tasks to the PositionDeleteFilter so that PositionDeleteFilter only needs to build a map with corresponding data files.